Recompiling lwip gives error

### Basic Infos

- [X ] This issue complies with the [issue POLICY doc](https://github.com/esp8266/Arduino/blob/master/POLICY.md).
- [X ] I have read the documentation at [readthedocs](https://arduino-esp8266.readthedocs.io/en/latest) and the issue is not addressed there.
- [X] I have tested that the issue is present in current master branch (aka latest git).
- [X ] I have searched the issue tracker for a similar issue.
- [X ] If there is a stack dump, I have decoded it.
- [X ] I have filled out all fields below.

#### Platform

- Hardware:					ESP8266
- Core Version:			[latest git ]
- Development Env:		[Arduino IDE]
- Operating System:		[Ubuntu]

### Settings in IDE

- Module:  			[Nodemcu]
- Flash Size: 		[4MB]
- lwip Variant:   [|v2 Lower Memory]
- Reset Method:		[nodemcu]
- Flash Frequency:	[40Mhz]
- CPU Frequency:		[80Mhz]
- Upload Using:		[SERIAL]
- Upload Speed:   [115200] (serial upload only)


### Problem Description

Recompiling the lwip library using following steps
1. Navigate to esp8266/tools/sdk/lwip/src
2. run make install command

### Error

app/dhcpserver.c:22:34: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'ICACHE_RODATA_ATTR'
 static const uint32 magic_cookie ICACHE_RODATA_ATTR = 0x63538263;
                                  ^
app/dhcpserver.c: In function 'create_msg':
app/dhcpserver.c:268:26: error: 'magic_cookie' undeclared (first use in this function)
   uint32 magic_cookie1 = magic_cookie;
                          ^
app/dhcpserver.c:268:26: note: each undeclared identifier is reported only once for each function it appears in
app/dhcpserver.c: In function 'parse_msg':
app/dhcpserver.c:559:14: error: 'magic_cookie' undeclared (first use in this function)
             &magic_cookie,
              ^
make: *** [build/app/dhcpserver.o] Error 1


Can you please suggest me how to resolve this error ?

--- a/tools/sdk/lwip/include/arch/cc.h
+++ b/tools/sdk/lwip/include/arch/cc.h
@@ -39,6 +39,7 @@
 #include "ets_sys.h"
 #include "osapi.h"
 #define EFAULT 14
+#include <sys/pgmspace.h>
 
 //#define LWIP_PROVIDE_ERRNO
 

